### PR TITLE
feat: add recipe details screen

### PR DESCRIPTION
**Pull request by abhishek-gkmit**
- add recipe details screen and files related to it
- add helper functions, api methods and data formatters related to it

### CODE REVIEW

File: src/screens/recepiDetails/RecepiDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import useThemeContext from '@hooks/useThemeContext';
+import { ImagePlaceholder } from '@constants/images';
+import { getRecipeDetails } from '@network/apiMethods';
+
+import RecipeDetailsTabView from './RecipeDetailsTabView';
+import {
+  RecipeDetailsContext,
+  initialContextValues,
+} from './RecipeDetailsContext';
+import getGlobalStyles from '@theme/globalStyles';
```
Comment (adiTrivsme) [inline src/screens/recepiDetails/RecepiDetails.tsx]: mixed import order

File: src/screens/recepiDetails/RecepiDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import recepiDetailsStyles from './styles';
+
+function BackAndFavouritesButtons({ navigation }: BackAndFavouritesButtonsProps) {
+  const { colors } = useThemeContext();
+
+  const localStyles = useStyles(recepiDetailsStyles);
+
+  return (
+    <FadeInSlideUp style={{ zIndex: 5 }}>
```
Comment (adiTrivsme) [inline src/screens/recepiDetails/RecepiDetails.tsx]: remove inline styles

File: src/screens/recepiDetails/RecepiDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+  const loadRecipeDetails = useCallback(async () => {
+    const recipeDetails = await getRecipeDetails(route.params?.recepiId);
+
+    recipeDetails && setRecipeDetails(recipeDetails);
+    setLoading(false);
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => setLoading(false), 500);
```
Comment (adiTrivsme) [inline src/screens/recepiDetails/RecepiDetails.tsx]: comment on the use of `setTimeout` here

File: src/screens/recepiDetails/RecepiDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
+import { useCallback, useEffect, useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import FastImage from 'react-native-fast-image';
+
+import TextBlock from '@components/customText';
+import FadeInSlideUp from '@components/fadeInSlideUp';
```
Comment (adiTrivsme) [inline src/screens/recepiDetails/RecepiDetails.tsx]: put all animations inside a sub-folder in the components folder

File: src/screens/recepiDetails/RecipeDetailsTabView.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+function RecipeInstructionsList() {
+  const { steps: instructions, servings } = useRecipeDetails();
+
+  const styles = useStyles(recepiDetailsStyles);
+
+  return (
+    <ScrollView contentContainerStyle={styles.recipeInstructionList}>
+      <View style={styles.recipeInstructionListHeading}>
+        <TextBlock>{servings + (servings > 1 ? ' serves' : ' serve')}</TextBlock>
```
Comment (adiTrivsme) [inline src/screens/recepiDetails/RecipeDetailsTabView.tsx]: extract to a variable

File: src/screens/recepiDetails/styles.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      padding: moderateScale(10),
+      elevation: 5,
+    },
+
+    favouriteButton: {
+      right: 0,
+      left: 'auto',
+    },
+
+    recipeDescription: { fontSize: 14, padding: 20 },
```
Comment (adiTrivsme) [inline src/screens/recepiDetails/styles.ts]: use fonts from constants and handle scaling

File: src/utility/helpers.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+export function getFraction(decimal: number) {
```
Comment (adiTrivsme) [inline src/utility/helpers.ts:3]: add comments to document the use and working of function

File: src/screens/recepiDetails/RecepiDetails.tsx (hunk 3 above)
Comment (abhishek-gkmit) [reply src/screens/recepiDetails/RecepiDetails.tsx]: It was there to simulate the loading screen for local testing purposes, forgot to remove it

**Review by adiTrivsme — APPROVED** [review]

**Pull request merged**